include/libcyphal/executor.hpp

```diff
@@ -29,164 +29,49 @@ using IExecutorTypeIdType = cetl::
 class IExecutor : public cetl::rtti_helper<IExecutorTypeIdType>
 {
 public:
-    /// @brief Defines a callback abstraction umbrella type.
+    /// NOSONAR cpp:S4963 - we do directly handle callback resource here.
     ///
-    struct Callback
+    class Callback
     {
-        /// @brief Defines low level unique identifier for a callback.
-        ///
-        /// 64 bits should be enough to ensure uniqueness of the callback identifier
-        /// by applying some simple rules (e.g. incrementing counter).
-        ///
-        using Id = std::uint64_t;
-
+    public:
         /// @brief Defines possible variants of callback schedules.
         ///
         struct Schedule
```

Comment:
It's a nit but maybe it doesn't need to be nested inside `Callback`? Fully qualified names look unwieldy with deep nesting.